lib/co_aspects/aspects/rescue_and_notify_aspect.rb

```diff
@@ -0,0 +1,65 @@
+require 'newrelic_rpm'
+
+module CoAspects
+  module Aspects
+    # Rescues any error and notifies NewRelic about it, without raising it
+    # again.
+    #
+    # Important: If CoAspects::Aspects::RescueAndNotifyAspect.enable_test_mode!
+    # is executed, then instead of calling NewRelic, it will raise the exception
+    # as if this aspect didn't exist. It's to be used by tests, to see
+    # exceptions being raised inside `_rescue_and_notify` annotated methods.
+    #
+    # Enabling test mode is not thread safe!
```

Comment:
Ideally not. Enabling test mode must be done explicitly and if it's done in the configuring process of RSpec, before tests are executed, then it will not be a problem.

It will certainly be a problem if someone is enabling or disabling test mode on a per test basis, but that doesn't make any sense on projects using the gem.